src/BoltOn.Data.CosmosDb/BaseCosmosDbRepository.cs

```diff
@@ -0,0 +1,118 @@
+            return entity;
+        }
+
+        public virtual async Task<TEntity> AddAsync(TEntity entity, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            await _client.CreateDocumentAsync(GetDocumentCollectionUri(), entity);
+            return entity;
+        }
+
+        public virtual IEnumerable<TEntity> FindBy(Expression<Func<TEntity, bool>> predicate, params Expression<Func<TEntity, object>>[] includes)
+        {
+            var query = _client.CreateDocumentQuery<TEntity>(GetDocumentCollectionUri())
+                .Where(predicate)
+                .AsDocumentQuery();
+
+            return GetResultsFromDocumentQuery(query).GetAwaiter().GetResult();
+        }
+
+        public virtual async Task<IEnumerable<TEntity>> FindByAsync(Expression<Func<TEntity, bool>> predicate,
+            CancellationToken cancellationToken = default(CancellationToken), params Expression<Func<TEntity, object>>[] includes)
+        {
+            var query = _client.CreateDocumentQuery<TEntity>(GetDocumentCollectionUri())
+               .Where(predicate)
+               .AsDocumentQuery();
+
+            return await GetResultsFromDocumentQuery(query);
+        }
+
+        public virtual IEnumerable<TEntity> GetAll()
+        {
+            var query = _client.CreateDocumentQuery<TEntity>
+                (GetDocumentCollectionUri())
+                .AsDocumentQuery();
+
+            return GetResultsFromDocumentQuery(query).GetAwaiter().GetResult();
+        }
+
+        public virtual async Task<IEnumerable<TEntity>> GetAllAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var query = _client.CreateDocumentQuery<TEntity>
+                (GetDocumentCollectionUri())
+                .AsDocumentQuery();
+
+            return await GetResultsFromDocumentQuery(query);
+        }
+
+        public virtual TEntity GetById<TId>(TId id)
+        {
+            var document = _client.ReadDocumentAsync<TEntity>(GetDocumentUri(id.ToString())).GetAwaiter().GetResult();
+            return document.Document;
+        }
+
+        public virtual async Task<TEntity> GetByIdAsync<TId>(TId id)
+        {
+            var document = await _client.ReadDocumentAsync<TEntity>(GetDocumentUri(id.ToString()));
+            return document.Document;
+        }
+
+        public virtual void Update(TEntity entity)
+        {
+            _client.UpsertDocumentAsync(GetDocumentCollectionUri(), entity).GetAwaiter().GetResult();
+        }
+
+        public virtual async Task UpdateAsync(TEntity entity, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            await _client.UpsertDocumentAsync(GetDocumentCollectionUri(), entity);
+        }
+
+        protected Uri GetDocumentCollectionUri()
+        {
+            return UriFactory.CreateDocumentCollectionUri(_databaseName, _collectionName);
```

Comment:
can't this be a property? and get initialized in the ctor, so that every time when this method gets called, the Uri will not be created